tiledb/sm/rest/rest_client_remote.cc

```diff
@@ -321,11 +323,12 @@ void RestClientRemote::post_array_from_rest(
     throw RestClientException("Error getting remote array; array is null.");
   }
 
-  Buffer buff;
+  SerializationBuffer buff{serialization_buffer_resource()};
   throw_if_not_ok(
-      serialization::array_open_serialize(*array, serialization_type_, &buff));
-  // Wrap in a list
-  BufferList serialized;
+      serialization::array_open_serialize(*array, serialization_type_, buff));
+  // Wrap in a list. Pass the buffer's allocator to avoid copying with the
+  // default allocator.
+  BufferList serialized{buff.get_allocator()};
```

Comment:
I am planning to get rid of the `BufferList` wrapper in a subsequent PR, by allowing to pass both single buffers and buffer lists (for queries) to `Curl`.